webui/react/src/ioTypes.ts

```diff
@@ -189,15 +189,23 @@ const ioDataLayer = io.type({
 
 const ioExpResources = io.type({ max_slots: optional(io.number) });
 
+const hParamTypes: Record<string, null> = Object
+  .values(ExperimentHyperParamType)
+  .reduce((acc, val) => ({ ...acc, [val]: null }), {});
+const ioHParamTypes = io.keyof(hParamTypes);
+const ioExpHParamVal = optional(io.union([ io.boolean, io.number, io.string ]));
```

Comment:
👍🏼  I'd make sure that the unit tests we have for decoding experiment config is actually hitting this new iotype

Reply:
added decoder tests to be able to handle non-primitive hps